File: 349-intersection-of-two-arrays/intersection-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    let set = new Set()
    let res = []
    nums1.sort((a, b) => a - b)
    nums2.sort((a, b) => a - b)
    let i = 0, k = 0
    while (i < nums1.length && k < nums2.length) {
        if (nums1[i] === nums2[k]) {
            if(res[res.length-1] !== nums1[i]) {
                res.push(nums1[i])
            }
            i++
            k++
    
        } else if (nums1[i] < nums2[k]) {
            i++
        } else k++
    }

    return res
};

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        seen = set(nums1)
        res = []

        for num in nums2:
            if num in seen:
                res.append(num)
                seen.remove(num)

        return res        